### ISSUE TITLE
You can't modify any ride in Scenario Editor, Roller Coaster Editor or in-game cheats

### ISSUE THREAD

**Issue by TheClaudeQC** [OpenRCT2/OpenRCT2#23504]
### Operating System

Windows 11

### OpenRCT2 build

OpenRCT2, v0.4.17 (25ac4cf5d6 on develop) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Describe the issue

so, whenever you dare to add any new rides/stall anywhere, it crashes (and I think the cultprit is the Fix of "closing the ride selection with the close window hotkey")

### Steps to reproduce

1. Open the Scenario Editor, Ride Conception or any map
2. try selecting a few new rides to add
3. GAME CRASHED

**Comment by sjoerddebruin** [comment]
Can confirm, the above describes the object selection window. @AaronVanGeffen, 25ac4cf seems to be the cause of this.

**Comment by AaronVanGeffen** [comment]
I would like to look into this, but cannot reproduce the crash. Does the game generate a crash dump file, by any chance?

**Comment by 733737** [comment]
can crash the game when changing water palette as well, but does not allow uploading to backtrace

**Comment by 733737** [comment]
[crash.zip](https://github.com/user-attachments/files/18272094/crash.zip)

**Comment by AaronVanGeffen** [comment]
There seems to be an awful lot of recursion going on here:
```
[...]
    frame #14765: 0x00000001006b64dc OpenRCT2`OpenRCT2::Ui::Windows::EditorObjectSelectionWindowCheck() + 212
    frame #14766: 0x00000001001156fc OpenRCT2`WindowCloseByClass(WindowClass) [inlined] WindowClose(w=0x000000012183fde0) at Window.cpp:230:12 [opt]
    frame #14767: 0x00000001001156ec OpenRCT2`WindowCloseByClass(WindowClass) at Window.cpp:258:13 [opt]
    frame #14768: 0x0000000100115698 OpenRCT2`WindowCloseByClass(cls=EditorObjectSelection) at Window.cpp:274:5 [opt]
    frame #14769: 0x00000001006b64dc OpenRCT2`OpenRCT2::Ui::Windows::EditorObjectSelectionWindowCheck() + 212
    frame #14770: 0x00000001001156fc OpenRCT2`WindowCloseByClass(WindowClass) [inlined] WindowClose(w=0x000000012183fde0) at Window.cpp:230:12 [opt]
    frame #14771: 0x00000001001156ec OpenRCT2`WindowCloseByClass(WindowClass) at Window.cpp:258:13 [opt]
    frame #14772: 0x0000000100115698 OpenRCT2`WindowCloseByClass(cls=EditorObjectSelection) at Window.cpp:274:5 [opt]
[...]
```

**Comment by AaronVanGeffen** [comment]
Fixed by #23511
